### PR TITLE
inference time monitor

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

File: mmdet/apis/ote/apis/detection/ote_utils.py [hunk 1]
```diff
@@ -106,3 +106,20 @@ def __init__(self, update_progress_callback: Optional[UpdateProgressCallback] =
     def on_train_batch_end(self, batch, logs=None):
         super().on_train_batch_end(batch, logs)
         self.update_progress_callback(self.get_progress())
+
+
+class InferenceProgressCallback(TimeMonitorCallback):
+    def __init__(self, num_test_steps, update_progress_callback: Optional[UpdateProgressCallback] = None):
```
Comment (druzhkov-paul) [inline mmdet/apis/ote/apis/detection/ote_utils.py]: Does `None` really work?

File: mmdet/apis/ote/apis/detection/ote_utils.py [hunk 2]
```diff
@@ -106,3 +106,20 @@ def __init__(self, update_progress_callback: Optional[UpdateProgressCallback] =
...
+        super().__init__(
+            num_epoch=0,
+            num_train_steps=0,
+            num_val_steps=0,
+            num_test_steps=num_test_steps,
+            update_progress_callback=update_progress_callback)
+
+    def on_test_batch_begin(self, batch=None, logs=None):
+        super().on_test_batch_begin(batch, logs)
```
Comment (druzhkov-paul) [inline mmdet/apis/ote/apis/detection/ote_utils.py]: Is it here for default values?

File: mmdet/apis/ote/apis/detection/task.py [hunk 3]
```diff
@@ -195,7 +203,8 @@ def infer(self, dataset: Dataset, inference_parameters: Optional[InferenceParame
 
     @staticmethod
     def _infer_detector(model: torch.nn.Module, config: Config, dataset: Dataset,
-                        eval: Optional[bool] = False, metric_name: Optional[str] = 'mAP') -> Tuple[List, float]:
+                        eval: Optional[bool] = False, metric_name: Optional[str] = 'mAP',
+                        time_monitor: InferenceProgressCallback = None) -> Tuple[List, float]:
```
Comment (druzhkov-paul) [inline mmdet/apis/ote/apis/detection/task.py]:
```suggestion
                        time_monitor: Optional[InferenceProgressCallback] = None) -> Tuple[List, float]:
```

File: mmdet/apis/test.py [hunk 4]
```diff
@@ -17,12 +17,15 @@ def single_gpu_test(model,
                     data_loader,
                     show=False,
                     out_dir=None,
-                    show_score_thr=0.3):
+                    show_score_thr=0.3,
+                    time_monitor=None):
```
Comment (druzhkov-paul) [inline mmdet/apis/test.py]: 😞

File: tests/test_ote_api.py [hunk 5]
```diff
@@ -21,6 +21,7 @@
 from ote_sdk.entities.shapes.ellipse import Ellipse
 from ote_sdk.entities.shapes.polygon import Polygon
 from ote_sdk.entities.task_environment import TaskEnvironment
+from ote_sdk.entities.inference_parameters import InferenceParameters
```
Comment (druzhkov-paul) [inline tests/test_ote_api.py]: Please sort.

File: mmdet/apis/test.py (hunk 4 above)
Comment (Ilya-Krylov) [reply mmdet/apis/test.py]: agree, but I don't see here another solution. What about you?

File: mmdet/apis/ote/apis/detection/ote_utils.py (hunk 2 above)
Comment (Ilya-Krylov) [reply mmdet/apis/ote/apis/detection/ote_utils.py]: yep

File: mmdet/apis/test.py (hunk 4 above)
Comment (druzhkov-paul) [reply mmdet/apis/test.py]: Seeing and thinking is generally not mine. But (quite random) guess is that `nn.Module.register_forward_hook` or `nn.Module.register_forward_pre_hook` might be suitable as well.

Comment (Ilya-Krylov) [reply mmdet/apis/test.py]: thanks for the hint. done

File: tests/test_ote_api.py (hunk 5 above)
Comment (Ilya-Krylov) [reply tests/test_ote_api.py]: sorted

**Review by druzhkov-paul — APPROVED** [review]

**Pull request merged**